test/e2e/cli_test.go

```diff
@@ -1733,6 +1735,138 @@ func (s *CLISuite) TestPluginStruct() {
 		})
 }
 
+func (s *CLISuite) TestTemplateLevelHooksWithRetry() {
+	var children []string
+	var workflowName string
+	var podName string
+	(s.Given().
+		Workflow(`
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  name: retries-with-hooks-and-artifact
+  labels:
+    workflows.argoproj.io/test: "true"
+  annotations:
+    workflows.argoproj.io/description: |
+      when retries and hooks are both included, the workflow cannot resolve the artifact 
+    workflows.argoproj.io/version: '>= 3.5.0'
+spec:
+  entrypoint: main
+  templates:
+    - name: main
+      steps:
+        - - name: build
+            template: output-artifact
+            hooks:
+              started:
+                expression: steps["build"].status == "Running"
+                template: started
+              success:
+                expression: steps["build"].status == "Succeeded"
+                template: success
+              failed:
+                expression: steps["build"].status == "Failed" || steps["build"].status == "Error"
+                template: failed
+        - - name: print
+            template: print-artifact
+            arguments:
+              artifacts:
+                - name: message
+                  from: "{{steps.build.outputs.artifacts.result}}"
+    
+    - name: output-artifact
+      script:
+        image: python:alpine3.6
+        command: [ python ]
+        source: |
+          import time
+          import random
+          import sys
+          time.sleep(random.randint(0,5)) # lifecycle hook for running won't trigger unless it runs for more than "a few seconds"
```

Comment:
`random.randint(0,5)` can return 0, this is a problem
I saw other templates in hooks_test.go uses sleep(1), is that sufficient?